tests/TestCase/View/ViewTest.php

```diff
@@ -646,6 +646,66 @@ public function testElement(): void
         $this->assertSame("Plugin element overridden in app\n", $result);
     }
 
+    /**
+     * Test fragment method with mock for element.
+     */
+    public function testFragmentMethod(): void
+    {
+        // Simulate the fragment storage with a variable
+        $storedFragment = '';
+
+        // Mock the element method to behave dynamically based on fragment data
+        $this->View = $this->getMockBuilder(View::class)
+            ->onlyMethods(['element'])
+            ->getMock();
+
+        // Set up the element method to return the stored fragment data
+        $this->View->expects($this->any())
+            ->method('element')
+            ->willReturnCallback(function ($path, $data) use (&$storedFragment) {
+                // Debug output to check what's happening
+                echo 'Data passed: ' . json_encode($data['data']) . "\n";
+                echo 'Stored fragment before update: ' . $storedFragment . "\n";
+
+                // If new fragment data is passed, update the stored fragment
+                if (!empty($data['data'])) {
+                    $storedFragment = $data['data'];
+                } elseif (!($data['data'] === null)) {
+                    $storedFragment = $data['data'];
+                }
+
+                // Debug output to confirm the change
+                echo 'Stored fragment after update: ' . $storedFragment . "\n";
+
+                // Return the stored fragment or an empty string if cleared
+                return $storedFragment;
+            });
+
+        // Step 1: Store initial fragment data
+        $this->View->fragment('testFragment', ['data' => 'Initial fragment data']);
+
+        // Verify the initial fragment content
+        $result = $this->View->fragment('testFragment');
+        echo 'Initial fragment: ' . $result . "\n";
+        $this->assertEquals('Initial fragment data', $result); // Expect stored data
```

Comment:
assertSame() should be used for scalar values

That said: Any such API change should go into 5.next instead of 5.x

Reply:
Thank you, I think it should be fixed now